api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/RegistrationServiceImpl.java

```diff
@@ -74,6 +86,29 @@ public void fetchFromNational(BiometricSubject nationalBiometricSubject, M2SysCa
         registerLocally(nationalBiometricSubject, fingerScan);
     }
 
+    @Override
+    public void synchronizeFingerprints(M2SysCaptureResponse fingerScan, FingerScanStatus fingerScanStatus) {
+        if (fingerScanStatus.isRegisteredNationally()) {
+            if (patientHelper.findByNationalFpId(fingerScanStatus.getNationalBiometricSubject().getSubjectId()) == null) {
+                attachNationalIdToThePatient(fingerScanStatus.getLocalBiometricSubject().getSubjectId());
+            }
+        } else {
+            registerNationally(fingerScanStatus.getLocalBiometricSubject(), fingerScan);
+        }
+    }
+
+    private void attachNationalIdToThePatient(String localId) {
+        Patient patient = patientHelper.findByLocalFpId(localId);
+
+        PatientIdentifierType patientIdentifierType = patientService.getPatientIdentifierTypeByUuid(
+                properties.getGlobalProperty(RegistrationCoreConstants.GP_BIOMETRICS_NATIONAL_PERSON_IDENTIFIER_TYPE_UUID));
+        Location location = locationService.getDefaultLocation();
```

Comment:
I think it is ok, but I have noticed that in some places (in i.e. registrationcore) the location is set null. @pgesek, what do you think about it?

Reply:
If this is the national id we can skip the id - but I'm not sure which id this s

Reply:
Location cannot be null in this case. The database won't accept that.